hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java

```diff
@@ -161,17 +159,115 @@ public void checkRunningRolloutsDoesNotStartNextGroupIfFinishConditionIsNotHit()
+        return createdRollout;
+    }
+
+    @Step("Finish three actions of the rollout group and delete two targets")
+    private void finishActionAndDeleteTargetsOfFirstRunningGroup(final Rollout createdRollout) {
+        // finish group one by finishing targets and deleting targets
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.RUNNING);
+        finishAction(runningActions.get(0));
+        finishAction(runningActions.get(1));
+        finishAction(runningActions.get(2));
+        targetManagement.deleteTargets(runningActions.get(3).getTarget().getId(),
+                runningActions.get(4).getTarget().getId());
+    }
+
+    @Step("Check the status of the rollout groups, second group should be in running status")
+    private void checkSecondGroupStatusIsRunning(final Rollout createdRollout) {
+        rolloutManagement.checkRunningRollouts(0);
+        // validate that the second group is in running state
+        final List<RolloutGroup> runningRolloutGroups = rolloutGroupManagement.findRolloutGroupsByRolloutId(
+                createdRollout.getId(), new OffsetBasedPageRequest(0, 10, new Sort(Direction.ASC, "id"))).getContent();
+        assertThat(runningRolloutGroups.get(0).getStatus()).isEqualTo(RolloutGroupStatus.FINISHED);
+        assertThat(runningRolloutGroups.get(1).getStatus()).isEqualTo(RolloutGroupStatus.RUNNING);
+        assertThat(runningRolloutGroups.get(2).getStatus()).isEqualTo(RolloutGroupStatus.SCHEDULED);
+    }
+
+    @Step("Finish one action of the rollout group and delete four targets")
+    private void finishActionAndDeleteTargetsOfSecondRunningGroup(final Rollout createdRollout) {
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.RUNNING);
+        finishAction(runningActions.get(0));
+        targetManagement.deleteTargets(runningActions.get(1).getTarget().getId(),
+                runningActions.get(2).getTarget().getId(), runningActions.get(3).getTarget().getId(),
+                runningActions.get(4).getTarget().getId());
+
+    }
+
+    @Step("Delete all targets of the rollout group")
+    private void deleteAllTargetsFromThirdGroup(final Rollout createdRollout) {
+        // delete all targets of the third group
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.SCHEDULED);
+        targetManagement.deleteTargets(runningActions.get(0).getTarget().getId(),
+                runningActions.get(1).getTarget().getId(), runningActions.get(2).getTarget().getId(),
+                runningActions.get(3).getTarget().getId(), runningActions.get(4).getTarget().getId());
+    }
+
+    @Step("Check the status of the rollout groups and the rollout")
+    private void checkStatusOfRolloutAndRolloutGroupsAfterDeletingTargets(final Rollout createdRollout) {
```

Comment:
What you are actually "check"? Maybe a better name would be `verifyRolloutAndAllGroupsAreFinished()`.